File: src/C#/CreationalPatterns/Builder/Laptops/ConcreteBuilders/ASUSConcreteBuilder.cs
using Laptops.Builder;
using Laptops.Product;
using System;
using System.Collections.Generic;
using System.Text;

namespace Laptops.ConcreteBuilders
{
    public class ASUSConcreteBuilder : LaptopBuilder
    {
        public ASUSConcreteBuilder()
        {
            Laptop = new Laptop();
        }

        public override void SetModel()
        {
            Console.Write("Enter Laptop Model: ");
            Laptop.Model = Console.ReadLine();
        }

        public override void SetCPUSeries()
        {
            Console.Write("Enter CPU Series: ");
            Laptop.CPUSeries = Console.ReadLine();
        }

        public override void SetGPUModel()
        {
            Console.Write("Enter GPU Model: ");
            Laptop.GPUModel = Console.ReadLine();
        }

        public override void SetRAMType()
        {
            Console.Write("Enter RAM Type: ");
            Laptop.RAMType = Console.ReadLine();
        }

        public override void SetRAMSize()
        {
            Console.Write("Enter RAM Size: ");
            Laptop.RAMSize = int.Parse(Console.ReadLine());
        }

        public override void SetDisplayType()
        {
            Console.Write("Enter Display Type: ");
            Laptop.DisplayType = Console.ReadLine();
        }

        public override void SetSSDType()
        {
            Console.Write("Enter SSD Type: ");
            Laptop.SSDType = Console.ReadLine();
        }

        public override void SetSSDCapacity()
        {
            Console.Write("Enter SSD Capacity: ");
            Laptop.SSDCapacity = int.Parse(Console.ReadLine());
        }

        public override void SetExtras()
        {
            Console.Write("Add extra item (Exit with END): ");
            string extraItem = Console.ReadLine();

            while (extraItem != "END")
            {
                Laptop.Extras.Add(extraItem);
                Console.Write("Add extra item (Exit with END): ");
                extraItem = Console.ReadLine();
            }
        }
    }
}

File: src/JavaScript/CreationalPatterns/Builder/Laptops/ConcreteBuilders/asusConcreteBuilder.js
const LaptopBuilder = require("../Builder/laptopBuilder");
const Laptop = require("../Product/laptop");

const prompt = require('prompt-sync')();

class ASUSConcreteBuilder extends LaptopBuilder {
    constructor() {
        super();
        this.laptop = new Laptop();
    }

    getLaptop() {
        return this.laptop;
    }

    setModel() {
        const model = prompt("Enter Laptop Model: ");
        this.laptop.model = model;
    }

    setCPUSeries() {
        const cpuSeries = prompt("Enter CPU Series: ");
        this.laptop.cpuSeries = cpuSeries;
    }

    setGPUModel() {
        const gpuModel = prompt("Enter GPU Model: ");
        this.laptop.gpuModel = gpuModel;
    }

    setRAMType() {
        const ramType = prompt("Enter RAM Type: ");
        this.laptop.ramType = ramType;
    }

    setRAMSize() {
        const ramSize = parseFloat(prompt("Enter RAM Size: "));
        this.laptop.ramSize = ramSize;
    }

    setDisplayType() {
        const displayType = prompt("Enter Display Type: ");
        this.laptop.displayType = displayType;
    }

    setSSDType() {
        const ssdType = prompt("Enter SSD Type: ");
        this.laptop.ssdType = ssdType;
    }

    setSSDCapacity() {
        const ssdCapacity = parseFloat(prompt("Enter SSD Capacity: "));
        this.laptop.ssdCapacity = ssdCapacity;
    }

    setExtras() {
        let extraItem = prompt("Add extra item (Exit with END): ");

        while (extraItem !== "END") {
            this.laptop.extras.push(extraItem);
            extraItem = prompt("Add extra item (Exit with END): ");
        }
    }
}

module.exports = ASUSConcreteBuilder;